### DISCUSSION TITLE
Windows Terminal v1.21.2701.0

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#17970] [Announcements]
Bug fixes! Bug fixes for all! Many of these were backported from Canary, and are also being released with the 1.22 Preview servicing update.

> [!NOTE]
> We have encountered a pretty annoying hanging issue on Windows 11 23H2, for which we have **paused the rollout of Terminal 1.21 Stable.**
> If you encounter a hang when resuming from sleep mode or unlocking your display, sorry! Relief is coming in the form of a Windows Update.

## Changes

* <kbd>Alt+Keypad Plus</kbd> is no longer eaten unless `EnableHexNumpad` is enabled in the registry (#17954)
* You can now configure the default input scope for the IME with an option on the **Startup** page or the JSON key `defaultInputScope` (enum `default` (default), `alphanumericHalfWidth`); this should allow you to force Terminal to accept English input even if you usually type in a different language (#17953)

## Bug Fixes

* You can now successfully input more than one Emoji at once on Windows 10 (#17213)
* In cmd and python (and other users of console line input), choosing a command from history will no longer insert it back into history a second time (#17852)
* Some localizations have been updated for correctness (#17958)
* Terminal has gotten much better about transferring focus to windows spawned by (and under, or above) hosted console applications (#17828)
* The search box will no longer cover up search results in the top right (unless there's nowhere to scroll) (#17516) (thanks @e82eric!)
* We will try much harder to keep your scroll position from changing when you have the search box open but are not looking at the results (#17885) (#17316)

### Accessibility

* The settings UI now responds more correctly to the systemwide text scaling setting (#17910)
* We've fixed the color contrast of the "reset" button in the settings page (#17912)

### Terminal Emulation and Console Compatibility

* ConPTY: responses longer than 4kb will no longer be corrupted (#17738) (thanks @j4james!)

### Reliability

* Closing split panes should crash roughly 50% less often (#17886)
* The experience of typing in and searching for multiple fonts has been (questionably?) improved (#17961)
* We should no longer crash while quitting (#17848)

---

_Binary files inside the unpackaged distribution archive bear the version number `1.21.240926001`._

### Asset Hashes

- Microsoft.WindowsTerminal_1.21.2701.0_8wekyb3d8bbwe.msixbundle_Windows10_PreinstallKit.zip
   - SHA256 `C498A534E9601273930C22A86483119613A4AF670DFB19FC016BABED1BE2FA8F`
- Microsoft.WindowsTerminal_1.21.2701.0_8wekyb3d8bbwe.msixbundle
   - SHA256 `ED7538F84002AE0C9C2B8D201FE4DE0CD6C3BAC56BF4D6CA9301B75DC4450A0F`
- Microsoft.WindowsTerminal_1.21.2701.0_arm64.zip
   - SHA256 `65D67C4ADD4AC09D2C34F9C2BC21C38AAF36BD72590B9FF51A0563D78B1B695A`
- Microsoft.WindowsTerminal_1.21.2701.0_x64.zip
   - SHA256 `2F712872ED7F552763F3776EA7A823C9E7413CFD5EC65B88E95162E93ACEF899`
- Microsoft.WindowsTerminal_1.21.2701.0_x86.zip
   - SHA256 `A8376386DD7AF6D3E124A97AC1996E3273179B5CCA6C4A75FC89C81960359C4A`

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.21.2701.0'>Windows Terminal v1.21.2701.0</a>.</em>

**Comment by thomaslevesque** [comment]
> We have encountered a pretty annoying hanging issue on Windows 11 23H2, for which we have paused the rollout of Terminal 1.21 Stable.
> If you encounter a hang when resuming from sleep mode or unlocking your display, sorry! Relief is coming in the form of a Windows Update.

I've been waiting for this release for a while, because it supposedly has a fix for a [bug](https://github.com/microsoft/terminal/issues/16077) that's been annoying me for many months. But I'm not getting the update via the Microsoft Store, probably because you paused the rollout.
Which Windows Update will fix it? Has it been rolled out already? Once it's installed, will the Terminal update become available?

**Reply by DHowett** [reply]
If you’re in a hurry to get the update, you can always install the msixbundle release artifact! It will continue to update via the store afterwards, so you aren’t risking disconnecting it from updates. You can jump the queue that way 😄

FWIW, we don’t have any automatic roll out set up for when the fix is available. I get to look at the reliability numbers and make a judgement call.

**Reply by thomaslevesque** [reply]
Cool, thanks! And I can still rollback if I encounter the issue, right?

**Reply by DHowett** [reply]
Absolutely. Just grab a copy of your `settings.json` and any files in the same directory. The new version may make some changes that aren’t forward compatible.

**Comment by dsolodow** [comment]
This hang issue looks to also impact 24H2; it was driving me **nuts** until I figured out it was Terminal rather than Windows.
Any information on the Windows update that will resolve this issue as I got 1.21 before it was paused.

**Reply by dsolodow** [reply]
Has this issue been resolved? Terminal 1.21.2911.0 on Windows 24H2 (26100.2161) appears to still be having the issue.

**Reply by DHowett** [reply]
Can you grab a dump of WindowsTerminal.exe when it's in this state? I don't think we've gotten any other reports (other than you - sorry to ignore you!) of it on 24H2. It may be a different issue.

For sharing dumps, my e-mail's in my profile. 🙂

**Comment by dsolodow** [comment]
What I run into may be different. Running terminal elevated (different user
account) and sometimes when a different uac prompt comes up I just get a
black screen with the pointer and it stays unresponsive until power cycle
